### PR TITLE
Fix duplicate records of inner instructions

### PR DESCRIPTION
**Pull request by jstarry**
#### Problem
Inner instructions are duplicately recorded for each instruction in a transaction due to an interior mutability bug.

#### Summary of Changes
- Fix bug and add test
- Deprecate bad inner instruction data in blockstore/bigtable
- Add new inner instruction field to metadata in blockstore/bigtable

Fixes https://github.com/solana-labs/solana/issues/13379

### CODE REVIEW

**Comment by jstarry** [comment]
Tested locally against blockstore and emulated bigtable.. old inner instructions are no longer returned and new ones are not duplicated. Protobuf worked as expected!

File: local-cluster/tests/local_cluster.rs [hunk 1]
```diff
@@ -1315,10 +1315,11 @@ fn test_fake_shreds_broadcast_leader() {
 fn test_faulty_node(faulty_node_type: BroadcastStageType) {
     solana_logger::setup();
     let num_nodes = 2;
-    let validator_config = ValidatorConfig::default();
     let mut error_validator_config = ValidatorConfig::default();
     error_validator_config.broadcast_stage_type = faulty_node_type;
-    let mut validator_configs = vec![validator_config; num_nodes - 1];
+    let mut validator_configs = Vec::with_capacity(num_nodes - 1);
+    validator_configs.resize_with(num_nodes - 1, ValidatorConfig::default);
```
Comment (jstarry) [inline local-cluster/tests/local_cluster.rs:1321]: This came up while I searched for other instances of the same bug. `ValidatorConfig` has interior mutability as well so I patched this up despite it not causing any issues for this test

**Comment by mvines** [comment]
Can we avoid the `bad_inner_instructions` field?   Reuse the existing file and either purge the inner instructions out of bigtable for older slots, or just don't care about the invalid data.

**Comment by jstarry** [comment]
Sure, we can a use a version field in the struct to ignore old bad data

**Comment by mvines** [comment]
> Sure, we can a use a version field in the struct to ignore old bad data

Or not even?  We can just document in an issue that there's bad inner instruction metadata data for slots < N on mainnet-beta.  If/when somebody actually cares, we can write/run a custom purge on BigTable directly to remove the data.

**Comment by jstarry** [comment]
Ok, that's fine. Whenever we update the warehouse nodes we should be sure to note the slot when this code is activated then.

**Comment by mvines** [comment]
> Ok, that's fine. Whenever we update the warehouse nodes we should be sure to note the slot when this code is activated then.

Happy to load up bw1 with a "stable" channel build as soon as this PR hits v1.3

**Pull request merged**